t/plugin/key-auth.t

```diff
@@ -215,3 +215,56 @@ apikey: auth-13
 ["passed\n", "hello world\n"]
 --- no_error_log
 [error]
+
+
+
+=== TEST 9: add consumer with empty key
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/consumers',
+                ngx.HTTP_PUT,
+                [[{
+                    "username": "error",
+                    "plugins": {
+                        "key-auth": {
+                        }
```

Comment:
> add consumer with empty key

should throw an error message if there is no argument when create a new consumer.

Reply:
I think we should not simply throw errors in the consumer, but have a mechanism for the plugin to define its own verification behavior

Reply:
Add a new parameter, which is the resource type to which the plugin is bound.

Determine the content of the authentication plug-in according to the type of resource created.

Here is an example of `key-auth`:

```lua
function _M.check_schema(conf, parrent_type)
    local ok, err = core.schema.check(schema, conf)
    if not ok then
        return false, err
    end

    if parrent_type == "consumer" then
        if core.table.nkeys(conf) == 0 then
            return false, "missing properties"
        end
    end

    return true
end
```

Reply:
I think it is good, but maybe we should have a discuss in mail list like the changes about core?

Reply:
nice, you can create a topic about this ^_^

Reply:
any update?